cmd/find-main.go

```diff
@@ -0,0 +1,190 @@
+		cli.StringFlag{
+			Name:  "older",
+			Usage: "Match all objects older than specified time in units (see UNITS)",
+		},
+		cli.StringFlag{
+			Name:  "larger",
+			Usage: "Match all objects larger than specified size in units (see UNITS)",
+		},
+		cli.StringFlag{
+			Name:  "smaller",
+			Usage: "Match all objects smaller than specified size in units (see UNITS)",
+		},
+		cli.StringFlag{
+			Name:  "maxdepth",
+			Usage: "Limit directory navigation to specified depth",
+		},
+		cli.BoolFlag{
+			Name:  "watch",
+			Usage: "Monitor specified location for newly created objects",
+		},
+		cli.BoolFlag{
+			Name:  "or",
+			Usage: "Changes the matching criteria from an \"and\" to an \"or\"",
+		},
+	}
+)
+
+var findCmd = cli.Command{
+	Name:   "find",
+	Usage:  "Finds files which match the given set of parameters.",
+	Action: mainFind,
+	Flags:  append(findFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} PATH FLAG EXPRESSION [FLAG]
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+   1. Find all files named foo from all buckets.
+   	$ {{.HelpName}} s3 -name "file"
+   
+   2. Find all text files from mybucket.
+   	$ {{.HelpName}} s3/mybucket -name "*.txt"
+   
+   3. Print only the object names without the directory component under this bucket.
+   	$ {{.HelpName}} s3/bucket -name "*" -print {base}
+   
+   4. Copy all jpg files from AWS S3 photos bucket to minio play test bucket.
+   	$ {{.HelpName}} s3/photos -name "*.jpg" -exec "mc cp {} play/test"
+
+   5. Find all jpg images from any folder prefixed with album.
+   	$ {{.HelpName}} s3/photos -name "*.jpg" -path "*/album*/*"
+   
+   6. Find all jpgs, pngs, and gifs using regex
+   	$ {{.HelpName}} s3/photos -regex "(?i)\.(jpg|png|gif)$"
+
+   7. Mirror all photos from s3 bucket *coninuously* from the s3 bucket to minio play test bucket.
+   	$ {{.HelpName}} s3/buck -name "*foo" -watch -exec "mc cp {} play/test"	
+
+   8. Generate self expiring urls (7 days), for all objects between 64 MB, and 1 GB in size. 
+   	$ {{.HelpName}} s3 -larger 64MB -smaller 1GB -print {url}
+
+   9. Find all files under the s3 bucket which were created within a week. 
+   	$ {{.HelpName}} s3/bucket -newer 1w
+
+   10. Find all files which were created more than 6 months ago ignoring files ending in jpg.
+   	$ {{.HelpName}} s3 -older 6m -ignore "*.jpg"
+
+   11. List all objects up to 3 levels subdirectory deep. 
+   	$ {{.HelpName}} s3/bucket -maxdepth 3
+
+FORMAT
+
+Smaller/Larger Flags: Will accept the suffixes: "b": Byte, "kib": KiByte, "kb": KByte, "mib": MiByte, "mb": MByte, "gib": GiByte, "gb": GByte, "tib": TiByte, "tb": TByte, "pib": PiByte, "pb": PByte, "eib": EiByte, "eb": EByte, // Without suffix "": Byte, "ki": KiByte, "k": KByte, "mi": MiByte, "m": MByte, "gi": GiByte, "g": GByte, "ti": TiByte, "t": TByte, "pi": PiByte, "p": PByte, "ei": EiByte, "e": EByte,
+
+   Older/Newer Flags: Will accept the suffixes d, w, m, y.
+
```

Comment:
This section does not look complete. It should probably be renamed as UNITS:
```
UNITS:
   --smaller, --larger flags accept human-readable case-insensitive number suffixes such as "k", "m", "g" and "t" referring to the metric units KB, MB, GB and TB respectively. Adding an "i" to these prefixes, uses the IEC units, so that "gi" refers to "gibibyte" or "GiB". A "b" at the end is also accepted. Without suffixes the unit is bytes.

   --older, --newer flags accept the suffixes "d", "w", "m" and "y" to refer to units of days, weeks, months and years respectively. <details about 1year=365 days or 12 months, etc> 
```